src/KafkaNET.Library/Consumers/PartitionTopicInfo.cs

```diff
@@ -218,6 +220,39 @@ public long FetchOffset
         // Gets or sets the offset for the next request.
         public long NextRequestOffset { get; internal set; }
 
+        public bool ConsumedOffsetOutOfRange
+        {
+            get
+            {
+                lock(consumedOffsetOutOfRangeLock)
+                {
+                    return this.consumedOffsetOutOfRangeFlag;
+                }
+            }
+            internal set
+            {
+                lock (consumedOffsetOutOfRangeLock)
+                {
+                    this.consumedOffsetOutOfRangeFlag = value;
+
+                    if (this.consumedOffsetOutOfRangeFlag)
+                    {
+                        // Clear the chunkQueue as current messages in the queue are invalid on brokers. Drop them in order to fetch valid messages in.
+                        while (this.chunkQueue.Count > 0)
+                        {
+                            FetchedDataChunk item = null;
+                            this.chunkQueue.TryTake(out item);
+                        }
+                        Logger.Info("Clearing the current data chunk due to consumedoffset out of range.");
```

Comment:
Can we place log message before clearing (where comment is), and maybe after as well, but with "Cleared" instead of Clearing